source.go

```diff
@@ -209,18 +209,17 @@ func (p *SourceObjectParams) AppendTo(body *form.Values, keyParts []string) {
 // but stored in JSON under a hash named after the `type` of the source.
 func (s *Source) UnmarshalJSON(data []byte) error {
 	type source Source
-	var ss source
-	err := json.Unmarshal(data, &ss)
-	if err != nil {
+	var v source
+	if err := json.Unmarshal(data, &v); err != nil {
 		return err
 	}
-	*s = Source(ss)
+	*s = Source(v)
 
 	var raw map[string]interface{}
-	err = json.Unmarshal(data, &raw)
-	if err != nil {
+	if err := json.Unmarshal(data, &raw); err != nil {
 		return err
 	}
+
 	if d, ok := raw[s.Type]; ok {
```

Comment:
the code at this line still kind of bothers me and feel not really Go agnostic I wonder if we should revisit at some point to list every single Source type explicitly with their own class/struct and such.

Reply:
Man, I was totally thinking exactly the same thing ...

Especially when we have things like balance transaction sources, etc. modeled to have one field per possible expansion type, this should really be the same thing.

I think the idea at the time is that there was still too many changes happening in sources to be able to model them accurately, but the API should be stable by now.